### PR TITLE
Register

### PR DESCRIPTION
**Pull request by kyokidG**
Sorry for mixing some fixing Login stuff with Register stuff ^^

### CODE REVIEW

File: app/auth/views.py [hunk 1]
```diff
@@ -15,11 +16,28 @@ def register():
...
         )
-        db.session.add(new_user)
-        db.session.commit()
-        return redirect(url_for('.login'))
+        try:
+            #when the user tries to register with a used username, SQLAlchemy throws a nasty error
+            db.session.add(new_user)
+            db.session.commit()
+        except exc.IntegrityError as e:
+            #catch that error and reload the register page, letting the user know that the username is already used
```
Comment (simutoni) [inline app/auth/views.py]: maybe you can flash a message here. see flask.flash

File: app/auth/views.py [hunk 2]
```diff
@@ -15,11 +16,28 @@ def register():
...
+            return render_template('auth/register.html',
+                           title="Register",
+                           form=form,
+                           used_username=True)
+        #after the new user has registered, I will log him in automatically
+        login_user(new_user)
+        return redirect(url_for('main.index'))
+        #old implementation required user to login after registration - which I think is annoying
```
Comment (simutoni) [inline app/auth/views.py]: I think you should delete this.

File: app/static/css/login.css [hunk 3]
```diff
@@ -7,26 +7,32 @@
     padding: 45px;
     text-align: center;
     box-shadow: 0 0 20px 0 rgba(0, 0, 0, 0.2), 0 5px 5px 0 rgba(0, 0, 0, 0.24);
+    height: 260px; /*button bug fixed - form was getting smaller as the button was pressed*/
```
Comment (simutoni) [inline app/static/css/login.css]: do not add comments with how things worked before.

File: app/static/css/login.css [hunk 4]
```diff
@@ -7,26 +7,32 @@
...
+    border: none;
+    margin: 0 0 20px 0;
+    outline: none;
+    color: #243C24;
+    text-align: center;
+    border-bottom: 1px solid #89E189;
+}
+.login-form input:focus {
+    margin: 0 0 18px 0; /*hackish solutin - when input is focused, bottom margin is 10px to make room for border*/
+    border-bottom: 3px solid #4B7B4B;
```
Comment (simutoni) [inline app/static/css/login.css]: please add a blank line between classes.

File: app/auth/views.py (hunk 1 above)
Comment (kyokidG) [inline app/auth/views.py]: Doesn't work with flash, I tried that.
Isn't ok to re-render the register html template with an extra parameter? ^^

File: app/auth/views.py [hunk 5]
```diff
@@ -1,6 +1,7 @@
-from flask import Blueprint, render_template, redirect, url_for
+from flask import Blueprint, render_template, redirect, url_for, flash
 from flask_login import login_user, logout_user, login_required
 from models import User
```
Comment (cristimihok) [inline app/auth/views.py]: This should come after third party imports

File: app/auth/views.py [hunk 6]
```diff
@@ -15,11 +16,22 @@ def register():
         pw_hash = bcrypt.generate_password_hash(form.password.data)
         new_user = User(
             form.username.data,
-            pw_hash
+            pw_hash,
```
Comment (cristimihok) [inline app/auth/views.py]: It is ok to use the whole word in this case "password"

File: app/auth/views.py [hunk 7]
```diff
@@ -15,11 +16,22 @@ def register():
...
         )
-        db.session.add(new_user)
-        db.session.commit()
-        return redirect(url_for('.login'))
+        try:
+            db.session.add(new_user)
+            db.session.commit()
+        except exc.IntegrityError as e:
+            return render_template('auth/register.html',
+                           title="Register",
```
Comment (cristimihok) [inline app/auth/views.py]: I think this is a pep8 issue. Parameters of a call should be indented at the same level.

File: app/static/css/login.css [hunk 8]
```diff
@@ -7,26 +7,32 @@
     padding: 45px;
     text-align: center;
     box-shadow: 0 0 20px 0 rgba(0, 0, 0, 0.2), 0 5px 5px 0 rgba(0, 0, 0, 0.24);
```
Comment (cristimihok) [inline app/static/css/login.css]: Try ordering css rules alphabetically. It is easier to find a specific rule.

File: app/static/css/login.css [hunk 9]
```diff
@@ -7,26 +7,32 @@
...
+    border-bottom: 1px solid #89E189;
+}
+
+.login-form input:focus {
+    margin: 0 0 18px 0;
+    border-bottom: 3px solid #4B7B4B;
 }
 
 .btn-login {
     width: 100% !important;
```
Comment (cristimihok) [inline app/static/css/login.css]: Isn't there another way to do this without "!important". It is highly discouraged to overrule specificity like this and it's error prone and hard to maintain.

File: app/static/css/register.css [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    border-bottom: 1px solid #89E189;
+}
+
+.register-form input:focus {
+    margin: 20px 3% 10px 3%; /*when input is focused, bottom margin is 10px to make room for border*/
+    border-bottom: 3px solid #4B7B4B;
+}
+
+.btn-register {
+    width: 95% !important;
```
Comment (cristimihok) [inline app/static/css/register.css]: Again, not a good idea.

**Pull request merged**